### DISCUSSION TITLE
About `use.data`?

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#134] [General]
Seems that `lazyStateGetterSetter` setter hasn't changed too much (I mean, I saw the same code on other side effect before) and I wanted to know how it is safe to call it from within the build if it still calls `api.rebuild()` (which gave me some problems before if IRC).

Thanks

**Comment by GregoryConrad** [comment]
Yes, the implementation of `lazyStateGetterSetter` is now identical to `use.data` and the rebuild can be safely called from within the same capsule's build. However, I'd recommend using `use.data` or `use.lazyData` instead, which are more clearly tailored toward updating state from within build.

**Reply by busslina** [reply]
> I'd recommend using use.data or use.lazyData instead

Yes, I'm gonna use `use.data`

**Reply by busslina** [reply]
> rebuild can be safely called from within the same capsule's

So, you changed some internals to that be possible?

**Reply by GregoryConrad** [reply]
Yes I changed up the internals
